OperationSpark/service-signups
# Smoke test setup

Resolves #45 

Sets up GitHub workflow to run a smoke test against the production service upon successful deployment.

### Test Steps
- Fetch the first available open Info Session from the Greenlight API endpoint
- Create a test Signup form and `POST` to this service's production Cloud Function HTTP trigger
  - Form includes a test SMS number from Twilio 
- Fetch expected SMS message from Twilio API
- Parse Info Link in SMS
  - Eg: [https://sms.operationspark.org/m/eyJ...](https://sms.operationspark.org/m/eyJ0ZW1wbGF0ZSI6IkluZm9TZXNzaW9uIiwiem9vbUxpbmsiOiJodHRwczovL3VzMDZ3ZWIuem9vbS51cy93Lzg3OTU0NjE2NzM1P3RrPVVlczIzbHpJeV82VHJXX1d5eXhMbnVzTWlXa3Q1amVWdFlheWdRTzR6SW8uRFFNQUFBQVVlb0R4bnhaRGFtOVhSRU52YjFSUVlUUnlkRVpaU0dGbVJuTm5BQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUEiLCJkYXRlIjoiMjAyMy0wMS0wOVQxODowMDowMFoiLCJuYW1lIjoiSGVucmkiLCJsb2NhdGlvblR5cGUiOiJIWUJSSUQiLCJsb2NhdGlvbiI6eyJuYW1lIjoiT3BlcmF0aW9uIFNwYXJrIiwibGluZTEiOiI1MTQgRnJhbmtsaW4gQXZlIiwiY2l0eVN0YXRlWmlwIjoiTmV3IE9ybGVhbnMsIExBIDcwMTE3IiwibWFwVXJsIjoiaHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS9tYXBzL3BsYWNlLzUxNCtGcmFua2xpbitBdmUlMkNOZXcrT3JsZWFucyUyQytMQSs3MDExNyJ9fQ==) 
- Checks URL shortener actually shortened the Info Link
  - (Eg: https://ospk.org/lztvCQIvUK) 
- Visit (short or long) link
- Ensure rendered page has expected content
  - [x] Session Date
  - [x] Session Time
  
  
  #### Additional Indo
 
  https://stackoverflow.com/questions/58457140/dependencies-between-workflows-on-github-actions
